jira_test.go

```diff
@@ -777,38 +780,40 @@ func TestSetOptionMixFunc(t *testing.T) {
 	assert := assert.New(t)
 
 	// reset command line arg
-	oldArgs := os.Args
-	defer func() { os.Args = oldArgs }()
-
-	os.Args = append(os.Args, "--token=123")
-	os.Args = append(os.Args, "--type=license")
-	os.Args = append(os.Args, "--assigneeId=654")
-	os.Args = append(os.Args, "--api=http://snyk.io/api")
-	os.Args = append(os.Args, "--configFile=./fixtures")
+	args := []string{
+		"--token=123",
+		"--type=license",
+		"--assigneeId=654",
+		"--api=http://snyk.io/api",
+		"--configFile=./fixtures",
+		"--orgID=456",
+		"--jiraProjectID=15699",
+	}
 
 	options := flags{}
-	options.setOption()
+	// consider explicitly passing CLI args into the fn, rather than relying on global variables (ie. os.Args)
+	options.setOption(args)
 
 	mandatoryResult := &MandatoryFlags{
 		orgID:         "456",
 		apiToken:      "123",
-		jiraProjectID: "15698",
+		jiraProjectID: "15699",
 		endpointAPI:   "http://snyk.io/api",
 	}
 
 	optionalResult := &optionalFlags{
-		assigneeID:             "654",
-		assigneeName:           "",
-		debug:                  false,
-		dryRun:                 false,
-		issueType:              "license",
-		jiraTicketType:         "Task",
-		labels:                 "",
-		maturityFilterString:   "proof-of-concept",
-		priorityIsSeverity:     true,
-		priorityScoreThreshold: 20,
-		projectID:              "",
-		severity:               "critical",
+		assigneeID: "654",
+		// 	assigneeName:           "",
+		// 	debug:                  false,
+		// 	dryRun:                 false,
+		issueType:      "license",
+		jiraTicketType: "Bug",
```

Comment:
I see that you changed the result here, but it should, this test is using both flags and config file and the jiraTicketType is set at Task in the config file, it should not be set to the default value, it the same with the severity.